contracts/simple_dex/lib.rs

```diff
@@ -477,10 +477,39 @@ mod simple_dex {
 
             Ok(())
         }
+        /// Returns the swap trade input given a desired amount and assuming a curve with equal token weights
+        ///
+        /// A_in = B_i * ((B_o / (B_o - A_o)) - 1)
```

Comment:
Question: This one ignores the swap fee - is it by design?
Suggestion: Is there any guarantee that if `out_given_in(x) = y`, then `in_given_out(y) = x`? I suspect there is going to be some inconsistency there, due to rounding. If the mismatch between the two is expected to be small, you could add a property test basically saying that if `out_given_in(x) = y`, then `in_given_out(y) = x +/- 1` or something.

Reply:
I think we will simply remove the fee altogether - since it will never be non-zero.

+1 on the test - I'm adding for a handful of PRs built on top of this branch to be merged and will most def add some more coverage

Reply:
Done, please take another look